pulsar-client-cpp/lib/ClientConnection.cc

```diff
@@ -933,23 +933,30 @@ ClientConnection::newConsumerStats(uint64_t consumerId, uint64_t requestId) {
         promise.setFailed(ResultNotConnected);
     }
     pendingConsumerStatsMap_.insert(std::make_pair(requestId, promise));
+    const SharedBuffer& buffer = Commands::newConsumerStats(outgoingCmd_, consumerId, requestId);
```

Comment:
This should not be a reference, because otherwise the object itself can get destroyed before you use it. Just make a regular copy (which is shallow copy, `SharedBuffer` is just having a `shared_ptr` internally). 

```cpp
SharedBuffer buffer = Commands... 
```